testing/integration/metrics_monitoring_test.go

```diff
@@ -0,0 +1,138 @@
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/elastic/elastic-agent-libs/kibana"
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools"
+	"github.com/elastic/elastic-agent/pkg/testing/tools/estools"
+)
+
+type MetricsRunner struct {
+	suite.Suite
+	info         *define.Info
+	agentFixture *atesting.Fixture
+
+	ESHost string
+}
+
+func TestMetricsMonitoringCorrectBinaries(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: Fleet,
+		Stack: &define.Stack{},
+		Local: false, // requires Agent installation
+		Sudo:  true,  // requires Agent installation
+		OS: []define.OS{
+			{Type: define.Linux},
+			{Type: define.Windows},
+		},
+	})
+
+	suite.Run(t, &MetricsRunner{info: info})
+}
+
+func (runner *MetricsRunner) SetupSuite() {
+	fixture, err := define.NewFixture(runner.T(), define.Version())
+	require.NoError(runner.T(), err)
+	runner.agentFixture = fixture
+
+	policyUUID := uuid.New().String()
+	basePolicy := kibana.AgentPolicy{
+		Name:        "test-policy-" + policyUUID,
+		Namespace:   "default",
+		Description: "Test policy " + policyUUID,
+		MonitoringEnabled: []kibana.MonitoringEnabledOption{
+			kibana.MonitoringEnabledLogs,
+			kibana.MonitoringEnabledMetrics,
+		},
+	}
+
+	unpr := false
+	installOpts := atesting.InstallOpts{
+		NonInteractive: true,
+		Force:          true,
+		Unprivileged:   &unpr,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+	defer cancel()
+
+	policyResp, err := tools.InstallAgentWithPolicy(ctx, runner.T(), installOpts, runner.agentFixture, runner.info.KibanaClient, basePolicy)
+	require.NoError(runner.T(), err)
+
+	_, err = tools.InstallPackageFromDefaultFile(ctx, runner.info.KibanaClient, "system", "1.53.1", "system_integration_setup.json", uuid.New().String(), policyResp.ID)
+	require.NoError(runner.T(), err)
+
+}
+
+func (runner *MetricsRunner) TestBeatsMetrics() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*20)
+	defer cancel()
+	agentStatus, err := runner.agentFixture.ExecStatus(ctx)
+	require.NoError(runner.T(), err)
+
+	componentIds := []string{
+		"system/metrics-default",
+		"log-default",
+		"beat/metrics-monitoring",
+		"elastic-agent",
+		"http/metrics-monitoring",
+		"filestream-monitoring",
+	}
+
+	require.Eventually(runner.T(), func() bool {
+		for _, cid := range componentIds {
+			query := genESQuery(agentStatus.Info.ID, cid)
+			res, err := estools.PerformQueryForRawQuery(ctx, query, "metrics-elastic_agent*", runner.info.ESClient)
+			require.NoError(runner.T(), err)
+			runner.T().Logf("Fetched metrics for %s, got %d hits", cid, res.Hits.Total.Value)
+			if res.Hits.Total.Value < 5 {
```

Comment:
Why `< 5`? Doesn't any amount of hits mean there was at least one document matching the query?

Reply:
Yah, my concern was that we could end up in some freak accident where a test misconfiguration causes us to re-use an agent install, and thus an agent ID, leading to some overlap of results. Not sure if this is realistic, though.

Reply:
The agent ID you get on enrollment is unique within Fleet, so shouldn't be possible unless Fleet is completely broken.

This isn't protecting against that properly anyway IMO, it is just a magic number. 

A way that should actually work would be:
1. Install and enroll an agent with monitoring disabled.
2. Wait one full metrics collection cycle.
3. Ensure there are no hits in the metrics-* indices for that agent.
4. Turn monitoring on.
5. Ensure there are metrics now.

I'm not sure if all of that is actually worth it though.